purescript-contrib/purescript-argonaut-codecs
# New tutorial for argonaut-codecs

## What does this pull request do?

Closes #48 and adds an in-depth tutorial to introduce new users to the library, provide them with the tools they need to perform common tasks, and includes a few problem situations along with their solutions at the end. I think it's critical that a new Argonaut user feels comfortable creating encoders and decoders for simple data types without having to look at tests, search for random examples across the web, or rely on Haskell or Scala documentation instead.

## Where should the reviewer start?

All changes are to the README.

## How should this be manually tested?

There is no new code to test, but if you'd like to try things out in the REPL and make sure there aren't errors, that would help; this also deserves an error-checking proofread.

To read the pretty, non-diff version of the file, use this link:
https://github.com/purescript-contrib/purescript-argonaut-codecs/blob/05d25c2ec89380dc7c6e8fdbc755150983bc0115/README.md

## Other Notes:

If I am missing any common scenarios that really ought to be covered by a tutorial, I am happy to add them in as well and am open to suggestions to improve this! They can also be subsequent pull requests.
